packages/liveblocks-node/src/thread-notification-helpers.ts

```diff
@@ -0,0 +1,284 @@
+import type {
+  BaseUserMeta,
+  CommentBody,
+  CommentBodyJson,
+  CommentData,
+  DRI,
+  DU,
+  OptionalPromise,
+  ResolveUsersArgs,
+} from "@liveblocks/core";
+import {
+  generateCommentUrl,
+  getMentionedIdsFromCommentBody,
+  transformCommentBody,
+} from "@liveblocks/core";
+
+import type { Liveblocks } from "./client";
+import type { ThreadNotificationEvent } from "./webhooks";
+
+export type ThreadNotificationCommentData = Omit<
+  CommentData,
+  "body" | "deletedAt"
+> & {
+  body: CommentBody;
+  deletedAt?: never;
+};
```

Comment:
If we need a type like this I would prefer that we export it from the same place as `CommentData` and call it more descriptively like `NonDeletedCommentData` or `ExistingCommentData`

Reply:
I did not wanted to have an impact on the current `CommentData` type definition. And I thought this definition would be used only for thread notification at first. 

WDYT of defining just a type named `CommentDataWithExistingBody` then?

Reply:
To me `ExistingCommentData` means the same as `CommentDataWithExistingBody` but shorter, but more generally maybe we don't even need a named type?

[Maybe letting TS narrowing on its own is enough?](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBDAnmYcDCEC2ngDsYAiAhjEXALxwDeAUHAssAFxwBEAxljvqwNx1wAlgBMWAZxhRBuAOb96MABZRgRYQElRcCVNny4UCFk3jJ0uQICuY4FBPazegexUlgwgIIwWxGMH3ugn6eMAD8Pm78AL5wAGRwABQCAD7UcABGEMKIprpycMLAADbAwV7hcLjAAG62cFEpaZnZFVW1ULwFxaXuXhF+9TQAlPw0AGaWuOwwghC4BRAAyliliuYA6kGKGNh4MAmcu-gsO9yEJERD1AJCY4mHZwB0hSVlMFe09PQqMJZQ87hLEUijcYsUbNcvgZSn95g89jc4AB6JFQuAAPVCiJRaIAckQoIYAO7uOAAA1Oe18RDJoJoUSAA)

Reply:
Ah sorry for me it's not the same 😅 
`ExistingCommentData` would mean in my understanding would mean that the all the data (or at least a large part) would exist. 

I do not think narrowing is a better idea here. We should be explicit enough to indicate the comment body is well existing and defined

Reply:
I was thinking on this and WDYT of creating a new type in `@liveblocks/core` named `ActiveCommentData` representing this:
```ts
export type ActiveCommentData = {
  type: "comment";
  id: string;
  threadId: string;
  roomId: string;
  userId: string;
  createdAt: Date;
  editedAt?: Date;
  reactions: CommentReaction[];
  body: CommentBody;
  deletedAt?: never;
};
```
I though about the word `active` as a comment a comment is active till it's not deleted.

Reply:
Not sure, but just to make sure regarding narrowing: it would be as explicit as the named version (you'll be free to do `comment.body` directly), just without a name (us struggling to find a name makes me think that the name we choose in the end won't be super clear to users too)

![](https://github.com/user-attachments/assets/4a7c4653-5a10-4c04-90e0-ccf751d18ea5)

Reply:
I ended up by doing this:
```tsx
import type { CommentBody, CommentData } from "../protocol/Comments";

export type CommentDataWithBody = Omit<CommentData, "body" | "deletedAt"> & {
  body: CommentBody;
  deletedAt?: never;
};

const isCommentDataWithBody = (
  comment: CommentData
): comment is CommentDataWithBody => {
  return comment.body !== undefined && comment.deletedAt === undefined;
};

export function filterCommentsWithBody(
  comments: CommentData[]
): CommentDataWithBody[] {
  const commentsWithBody: CommentDataWithBody[] = [];
  for (const comment of comments) {
    if (isCommentDataWithBody(comment)) {
      commentsWithBody.push(comment);
    }
  }
  return commentsWithBody;
}
```
LMK what you think